EIPS/eip-draft_held_token.md

```diff
@@ -0,0 +1,228 @@
+   * @param tokenAddress held token address
+   * @param owner functional token owner
+   * @return held token balance
+   */
+  function heldBalanceOf(address tokenAddress, address owner)
+    external
+    view
+    returns (uint256);
+}
+```
+
+Smart contracts implementing the `ERC1155` held token standard MUST implement all of the functions in the `IERC1155Holder` interface.
+
+```solidity
+/**
+ * @notice the ERC1155 holder standard provides a common interface to query
+ * token balance information
+ */
+interface IERC1155Holder is IERC165 {
+  /**
+   * @notice emitted when the token is transferred to the contract
+   * @param owner functional token owner
+   * @param tokenAddress held token address
+   * @param tokenId held token ID
+   * @param tokenAmount held token amount
+   */
+  event Hold(
+    address indexed owner,
+    address indexed tokenAddress,
+    uint256 indexed tokenId,
+    uint256 tokenAmount
+  );
+
+  /**
+   * @notice emitted when the token is released back to the user
+   * @param owner functional token owner
+   * @param tokenAddress held token address
+   * @param tokenId held token ID
+   * @param tokenAmount held token amount
+   */
+  event Release(
+    address indexed owner,
+    address indexed tokenAddress,
+    uint256 indexed tokenId,
+    uint256 tokenAmount
+  );
+
+  /**
+   * @notice get the held balance of the token owner
+   * @param tokenAddress held token address
+   * @param owner functional token owner
+   * @param tokenId held token ID
+   * @return held token balance
+   */
+  function heldBalanceOf(
+    address tokenAddress,
+    address owner,
+    uint256 tokenId
+  ) external view returns (uint256);
+}
+```
+
+## Rationale
+
+This interface is designed to be extremely lightweight and compatible with any existing token contract. Any token holder contract likely already stores all relevant information, so this standard is purely adding a common interface to expose that data.
+
+The token address parameter is included to support contracts that can hold multiple token contracts simultaneously. While some contracts may only hold a single token address, this is more general to either scenario.
+
+Separate interfaces are proposed for each token type (ERC20, ERC721, ERC1155) because any contract logic to support holding these different tokens is likely independent. In the scenario where a single contract does hold multiple token types, it can simply implement each appropriate held token interface.
+
+
+## Backwards Compatibility
+
+Importantly, the proposed specification is fully compatible with all existing ERC20, ERC721, and ERC1155 token contracts. 
+
+Token holder contracts will need to be updated to implement this lightweight interface.
+
+Consumer of this standard will need to be updated to respect this interface in any relevant ownership logic.
+
+
+## Reference Implementation
+
+A full example implementation can be found here:
+
+https://github.com/devinaconley/token-hold-example
```

Comment:
Please replace this external link with a _minimal_ reference implementation under `../assets/eip-4987`.

Reply:
fixed